rest/src/main/groovy/whelk/rest/api/ESQuery.groovy

```diff
@@ -79,20 +89,62 @@ class ESQuery {
         aggQuery = getAggQuery(queryParameters)
         filters = getFilters(queryParameters)
 
-        queryParameters = maybeResetTypeParam(queryParameters, originalTypeParam)
+        if (originalTypeParam != null) {
+            queryParameters.put('@type', originalTypeParam)
+        }
 
-        Map queryString = [
+        Map simpleQuery = [
             'simple_query_string': [
                 'query': q,
                 'default_operator':  'AND'
             ]
         ]
 
+        Map queryParam = simpleQuery
```

Comment:
`queryParam` is easy to confuse with `queryParameters` and so on. Maybe `queryClauses` is a better name?

Reply:
Much better; will change!